pkg/io/read.go

```diff
@@ -1,10 +1,16 @@
 package io
 
-import "io"
+import (
+	"fmt"
+	"io"
+)
 
 // ReadAllWithEstimate is a fork of https://go.googlesource.com/go/+/go1.16.3/src/io/io.go#626
 //  with a starting buffer size. if none is provided it uses the existing default of 512
 func ReadAllWithEstimate(r io.Reader, estimatedBytes int64) ([]byte, error) {
+	if estimatedBytes < 0 {
+		return nil, fmt.Errorf("negative value received for estimatedBytes")
```

Comment:
might be better just to use the 512?

Reply:
Other than the local backend which can return a negative value from the `stat()` no other backend/cache would return a negative, I _think_ an error here might be fine. But I wouldn't mind changing this to a warning log + just using the default 512 bytes. Let me know!

Reply:
+1 with using default instead of erroring, as the parameter is a performance hint not a strict requirement.